### PR TITLE
Tests config_homepage_text

### PR DESCRIPTION
**Pull request by troublemagnet**

### CODE REVIEW

File: project_admin/tests/tests.py [hunk 1]
```diff
@@ -43,3 +43,174 @@ def test_admin_login_fail(self):
...
         self.assertEqual(response.status_code, 200)
         self.assertContains(response, "Password incorrect")
+
+    def test_config_homepage_text_logged_out(self):
+        """
+        Test config_homepage_text function when logged out
+        """
+        c = Client()
+        response = c.get("/project-admin/config-homepage-text/")
+        self.assertEqual(response.status_code, 302)
```
Comment (gedankenstuecke) [inline project_admin/tests/tests.py]: Please use `self.assertRedirects` to check for redirects. You can use it like this: `self.assertRedirects(response, '/where-does-it-redirect-to/')`

File: project_admin/tests/tests.py [hunk 2]
```diff
@@ -43,3 +43,174 @@ def test_admin_login_fail(self):
...
+        self.assertEqual(response.context['config'].upload_description,
+                         'foo_bar')
+
+    def test_config_file_settings_logged_out(self):
+        """
+        Test config_file_settings function when logged out
+        """
+        c = Client()
+        response = c.get("/project-admin/config-file-settings/")
+        self.assertEqual(response.status_code, 302)
```
Comment (gedankenstuecke) [inline project_admin/tests/tests.py]: Should also use `self.assertRedirects` as it not only tests the status code but also the target.

File: project_admin/tests/tests.py [hunk 3]
```diff
@@ -43,3 +43,174 @@ def test_admin_login_fail(self):
...
+        self.assertTemplateUsed(response,
+                                'project_admin/config-file-settings.html')
+
+    def test_config_oh_settings_logged_out(self):
+        """
+        Test config_oh_settings function when logged out
+        """
+        c = Client()
+        response = c.get("/project-admin/config-oh-settings/")
+        self.assertEqual(response.status_code, 302)
```
Comment (gedankenstuecke) [inline project_admin/tests/tests.py]: `self.assertRedirects`

File: project_admin/tests/tests.py [hunk 4]
```diff
@@ -43,3 +43,174 @@ def test_admin_login_fail(self):
...
+        self.assertTemplateUsed(response,
+                                'project_admin/config-oh-settings.html')
+
+    def test_config_general_settings_logged_out(self):
+        """
+        Test config_general_settings function when logged out
+        """
+        c = Client()
+        response = c.get("/project-admin/config-general-settings/")
+        self.assertEqual(response.status_code, 302)
```
Comment (gedankenstuecke) [inline project_admin/tests/tests.py]: `self.assertRedirects` :)

**Review by gedankenstuecke — CHANGES_REQUESTED** [review]
That looks good overall! Instead of only testing for the redirect status code please also test the redirect target, both can be done in one go with `self.assertRedirects` (see inline comments)

**Comment by troublemagnet** [comment]
@gedankenstuecke Made the changes! Ready for review

**Review by gedankenstuecke — APPROVED** [review]

**Comment by gedankenstuecke** [comment]
That's great, thanks so much!

**Pull request merged**